dotcom-rendering/scripts/gen-stories/gen-stories.js

```diff
@@ -0,0 +1,171 @@
+/*
+
+Use: node dotcom-rendering/scripts/gen-stories/gen-stories.js
+
+This script was created as a replacement for storiesOf to generate all of the possible variants
+of our Card and Layout components.
+
+It should be run whenever any of the Display, Design, or Theme `format` properties change
```

Comment:
should it be run before the `chromatic ` script in the root `package.json`? that way you never need to remember to run it.

Reply:
so something like 

```json
	"chromatic": "node dotcom-rendering/scripts/gen-stories/gen-stories.js && chromatic --build-script-name=build-storybook --exit-zero-on-changes",
```

Reply:
or maybe better in `.storybook/main.js`, just before the config is exported

Reply:
> it might be useful to run this script from .storybook/main.js, just before the config is exported

If this triggers this code to run prior to the build then that sounds ideal. I find commit hooks a bit flakey personally

Reply:
It should recreate the stories every time storybook builds, I *think*

Reply:
I think I can do this, but I might have to make the script synchronous.

Reply:
> anyway we should just not check in the files

I think I'd say we should check them in - only for the purpose of when somebody makes a change to the gen-stories script, its visible in the git diff exactly how the output script has changed.

Reply:
Probably for outside of this PR

Just thinking about the dev flow for this.
- [`libs`](https://github.com/guardian/libs) updates something in the [Format](https://github.com/guardian/libs/blob/main/src/format/README.md)
- update `dotcom-rendering/package.json` to new version of `libs`
- commit
- push

Would you have expected to run `storybook` because you updated `libs`?

This also means you need to remember to do that, or remember to run `make gen-stories`, which adds a little ambiguity as to when to use which method.

By adding a build step to the start script, we also have potential to lose idempotence across environments. e.g. I don't check in the new auto-generated stories files, when story book is run, it will run with the updated files. This feels quite confusing.

Not sure I have a suggestion just yet, but thought I'd point out where I find this current setup confusing.